cibuildwheel/util.py

```diff
@@ -562,6 +565,46 @@ def virtualenv(
     return env
 
 
+@overload
+def find_abi3_wheel(wheels: Sequence[Path], identifier: str) -> Optional[Path]:
+    ...
+
+
+@overload
+def find_abi3_wheel(wheels: Sequence[PurePath], identifier: str) -> Optional[PurePath]:
+    ...
+
+
+def find_abi3_wheel(
+    wheels: Sequence[Union[Path, PurePath]], identifier: str
+) -> Optional[Union[Path, PurePath]]:
```

Comment:
It would be good to add a docstring here detailing what it's doing. e.g. 

```
'''
Finds an ABI3 wheel in `wheels` compatible with the Python
interpreter specified by `identifier`. 
'''
```